content/post/02-gitlab-alpinelinux-v4-api.md

```diff
@@ -0,0 +1,195 @@
+---
+title: "Contributing to Alpine Linux: making Merge Requests"
+date: 2019-10-29T11:28:05-03:00
+draft: false
+---
+
+# Introduction
+
+Some time ago, [Alpine Linux](https://alpinelinux.org) started using their self-hosted
+[GitLab instance](https://gitlab.alpinelinux.org) to accept contributions via merge requests.
+
+I wanted to move to it immediately as using their GitLab instance is planned to be the main repo (it
+is currently mirrors git.alpinelinux.org).
+
+# How it was before
+
+Before using their GitLab instance i used their [GitHub repo](https://github.com/alpinelinux/aports) to
+contribute, i would use the amazing [hub](https://github.com/github/hub) tool, which after login with my
+personal access token would allow me to easily check out pull requests and merge them from the CLI.
+
+The scripts for it are very simple, most of the logic is on hub and a few lines of shell just wrap around
+it for my most common use cases.
+
+This one uses hub to checkout a PR someone has made:
+
+```sh
+#!/bin/sh
+set -eu
+while [ $# -gt 0 ]
+do
+	if [ $# -gt 1 ]
+	then
+		args="$1 $2"
+		shift 2
+	else
+		args="$1"
+		shift 1
+	fi
+	hub pr checkout $args
+	pullp
+done
+```
+
+# Moving to GitLab
+
+GitLab doesn't have a tool like hub that makes it super easy to interact with the GitLab V4 API like
+hub does with the GitHub API. So i turned into accessing the API by myself with the ever trustworthy curl.
+
+## Finding the endpoint
+
+Since GitLab allows self hosting, first thing you have to do is find our own GitLab endpoint. GitHub
+is a centralized proprietary service so it doesn't have this issue.
+
+This is the following code present in all my scripts that deal with GitLab (i believe the comments
+are enough to tell what they are for and what they meant).
```

Comment:
` ` ` 
The following code is present in all my scripts that deal with GitLab (I believe the comments
are enough to tell what they are for and what they mean).
` ` `